worlds/soe/__init__.py

```diff
@@ -51,41 +64,75 @@
     pyevermizer.CHECK_NPC: _id_base + 400,  # npc 64400..64499
     # TODO: sniff 64500..64799
+    pyevermizer.CHECK_TRAP: _id_base + 900,  # npc 64900..64999
 }
 
 # cache native evermizer items and locations
 _items = pyevermizer.get_items()
+_traps = pyevermizer.get_traps()
 _locations = pyevermizer.get_locations()
 # fix up texts for AP
 for _loc in _locations:
     if _loc.type == pyevermizer.CHECK_GOURD:
         _loc.name = f'{_loc.name} #{_loc.index}'
+# item helpers
+_ingredients = (
+    'Wax', 'Water', 'Vinegar', 'Root', 'Oil', 'Mushroom', 'Mud Pepper', 'Meteorite', 'Limestone', 'Iron',
+    'Gunpowder', 'Grease', 'Feather', 'Ethanol', 'Dry Ice', 'Crystal', 'Clay', 'Brimstone', 'Bone', 'Atlas Amulet',
+    'Ash', 'Acorn'
+)
+_other_items = (
+    'Call bead', 'Petal', 'Biscuit', 'Pixie Dust', 'Nectar', 'Honey', 'Moniez'
+)
+
+
+def _match_item_name(item, substr: str) -> bool:
+    sub = item.name.split(' ', 1)[1] if item.name[0].isdigit() else item.name
+    return sub == substr or sub == substr+'s'
 
 
 def _get_location_mapping() -> typing.Tuple[typing.Dict[str, int], typing.Dict[int, pyevermizer.Location]]:
     name_to_id = {}
     id_to_raw = {}
     for loc in _locations:
-        apid = _id_offset[loc.type] + loc.index
-        id_to_raw[apid] = loc
-        name_to_id[loc.name] = apid
+        ap_id = _id_offset[loc.type] + loc.index
+        id_to_raw[ap_id] = loc
+        name_to_id[loc.name] = ap_id
     name_to_id['Done'] = None
     return name_to_id, id_to_raw
 
 
 def _get_item_mapping() -> typing.Tuple[typing.Dict[str, int], typing.Dict[int, pyevermizer.Item]]:
     name_to_id = {}
     id_to_raw = {}
-    for item in _items:
+    for item in itertools.chain(_items, _traps):
         if item.name in name_to_id:
             continue
-        apid = _id_offset[item.type] + item.index
-        id_to_raw[apid] = item
-        name_to_id[item.name] = apid
+        ap_id = _id_offset[item.type] + item.index
+        id_to_raw[ap_id] = item
+        name_to_id[item.name] = ap_id
     name_to_id['Victory'] = None
     return name_to_id, id_to_raw
 
 
+def _get_item_grouping() -> typing.Dict[str, typing.Set[str]]:
+    groups = {}
+    ingredients_group = set()
+    for ingredient in _ingredients:
+        group = set(item.name for item in _items if _match_item_name(item, ingredient))
+        groups[f'Any {ingredient}'] = group
+        ingredients_group |= group
+    groups['Ingredients'] = ingredients_group
+    for other in _other_items:
+        groups[f'Any {other}'] = set(item.name for item in _items if _match_item_name(item, other))
+    groups['Alchemy'] = set(item.name for item in _items if item.type == pyevermizer.CHECK_ALCHEMY)
+    groups['Weapons'] = {'Spider Claw', 'Horn Spear', 'Gladiator Sword', 'Bronze Axe', 'Bronze Spear', 'Crusader Sword',
+                         'Lance (Weapon)', 'Knight Basher', 'Atom Smasher', 'Laser Lance'}
+    groups['Bazooka'] = {'Bazooka+Shells / Shining Armor / 5k Gold'}
```

Comment:
this looks like "," was intended, not /?

Reply:
that's what the drop is called, because it (can) change depending on what you aleady have.